src/AsyncInitializationWithSimpleInjectorDemo.IntegrationTests/_Infrastructure/IntegrationTestWebApplicationFactory.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace AsyncInitializationWithSimpleInjectorDemo.IntegrationTests._Infrastructure
+{
+	public class IntegrationTestWebApplicationFactory : WebApplicationFactory<Startup>
```

Comment:
**Here is the important class!**  This class performs setup for the web host used for integration tests.  It is responsible for building the integration-test-specific configuration and for performing asynchronous initialization once the test server has been set up.